src/main/java/wolox/training/services/servicesImpl/UserServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package wolox.training.services.servicesImpl;
+
+import org.json.JSONException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+import wolox.training.services.OpenLibraryService;
+import wolox.training.services.UserService;
+
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    OpenLibraryService openLibraryService;
+
+    @Autowired
+    BookRepository bookRepository;
+
+    public User addBook(Long userId, Book bookToAdd) throws JSONException {
+        Optional<User> optional = userRepository.findById(userId);
+        Book book = openLibraryService.findByIsbn(bookToAdd.getIsbn());
+        if(optional.isPresent() && book != null){ // controller validates that user always exists.
```

Comment:
Again, take advantage of the orElseThrow structure for both user and book.

Reply:
I don't think it's appropiate to use it since `openLibraryService.findByIsbn` has logic to throw exceptions on it's own...